src/utils.js

```diff
@@ -1,7 +1,42 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import Loader from './Components/Loader/Loader';
+
 class Util {
     static sid() {
         return (1 + Math.random() * 429496729125).toFixed().toString();
     }
+
+    static withData(WrappedComponent) {
+        return (props) => {
+            const [myData, setMyData] = useState({});
+            // eslint-disable-next-line react/prop-types
+            const { promiseData, children, ...newProps } = props;
+
+            useEffect(() => {
+                promiseData().then((data) => {
+                    setMyData(data);
+                });
+            }, []);
+
+            newProps.data = myData;
+
+            if (newProps.data.length) {
+                return (
+                    <WrappedComponent {...newProps}>
+                        {children}
+                    </WrappedComponent>
+
+                );
+            }
+            return (<Loader />);
+        };
+    }
 }
 
+Util.propTypes = {
+    promiseData: PropTypes.objectOf(PropTypes.shape).isRequired,
+    children: PropTypes.objectOf(PropTypes.shape).isRequired,
+};
+
 export default Util;
```

Comment:
The problem with the meaningful name of Component will be solved exporting the static functions themselves but we can create a component to withData function and change the name. I think that will give some clarity to code and will delegate responsabilities.

About duplicated code instead helper, I think that helper is better because futere fixes and improves will be concentrated in a single place.